src/kernel.cr

```diff
@@ -2,12 +2,20 @@
   STDIN  = IO::FileDescriptor.new(0)
   STDOUT = IO::FileDescriptor.new(1).tap { |f| f.flush_on_newline = true }
   STDERR = IO::FileDescriptor.new(2).tap { |f| f.flush_on_newline = true }
+
+  ORIGINAL_STDIN  = STDIN
+  ORIGINAL_STDOUT = STDOUT
+  ORIGINAL_STDERR = STDERR
 {% else %}
   require "c/unistd"
 
   STDIN  = IO::FileDescriptor.from_stdio(0)
   STDOUT = IO::FileDescriptor.from_stdio(1).tap { |f| f.flush_on_newline = true }
   STDERR = IO::FileDescriptor.from_stdio(2).tap { |f| f.flush_on_newline = true }
+
+  ORIGINAL_STDIN  = IO::FileDescriptor.new(0, blocking: true)
+  ORIGINAL_STDOUT = IO::FileDescriptor.new(1, blocking: true).tap { |f| f.flush_on_newline = true }
+  ORIGINAL_STDERR = IO::FileDescriptor.new(2, blocking: true).tap { |f| f.flush_on_newline = true }
```

Comment:
I'm not sure. This creates more constants for a single use: reopen stdio file descriptors on fork/exec for POSIX targets. I think it was better placed in `Process#exec_internal`.

Given the refactor, can we just init on-demand in `Process.to_real_fd`? With `Process#exec_internal` just passing `STDIN`, ... to `reopen_io`? That would then be abstracted to be system specific, with `to_real_fd` being a noop on windows (unless it needs something):

```crystal
protected def self.exec_internal(...) : NoReturn
  reopen_io(input, STDIN)
  reopen_io(output, STDOUT)
  reopen_io(error, STDERR)

  # ...
end

private def self.to_real_fd(fd)
  {% unless flag?(:windows) %}
    # reopen original stdio devices on POSIX systems:
    case fd
    when STDIN then IO::FileDescriptor(0, blocking: true)
    when STDOUT then IO::FileDescriptor(1, blocking: true)
    when STDERR then IO::FileDescriptor(2, blocking: true)
  {% end %}
  fd
end
```

Reply:
Are there other use cases than restoring the original file descriptors on fork/exec for POSIX targets? which must be achieved through Process#fork and Process#exec.

I'd let them in `#to_real_fd`, at least for the time being, and see with use-cases (if any) and after integrating windows (that doesn't need it) if it's worth changing.

Reply:
Yeah I'll move these constants to private in `Process`.

Reply:
Done.